aiohttp/client_ws.py

```diff
@@ -252,17 +255,17 @@ async def receive(self, timeout: Optional[float] = None) -> WSMessage:
                 return WS_CLOSED_MESSAGE
 
             try:
-                self._waiting = self._loop.create_future()
+                self._waiting = True
                 try:
                     async with async_timeout.timeout(
                         timeout or self._timeout.ws_receive
                     ):
                         msg = await self._reader.read()
                     self._reset_heartbeat()
                 finally:
-                    waiter = self._waiting
-                    self._waiting = None
-                    set_result(waiter, True)
+                    self._waiting = False
+                    if close_wait := self._close_wait:
+                        set_result(close_wait, None)
```

Comment:
I feel like it's more readable without the assignment.
```suggestion
                    if self._close_wait:
                        set_result(self._close_wait, None)
```

Reply:
I usually use this pattern with futures to ensure we hold the future in case it gets unset inside the block if their is an await.  Its a learned pattern to avoid mistakes. It doesn't matter here though since execution is synchronous and there is no await. So let's not use the walrus here.